Assets/Scripts/PowerUps/SpeedPowerUp.cs

```diff
@@ -1,21 +1,33 @@
-﻿using System.Collections;
-using System.Collections.Generic;
-using UnityEngine;
+﻿using UnityEngine;
 
-public class SpeedPowerUp : MonoBehaviour
+[DisallowMultipleComponent]
+public class SpeedPowerUp : PowerUpBase
 {
-    void OnTriggerEnter(Collider c)
+    [SerializeField]
+    private const float m_SpeedModifier = 1.50f;
+
+
+    public override void OnPowerUpExpired()
     {
-        if (c.CompareTag("Player"))
+        PlayerMovement playerMovement = GetComponent<PlayerMovement>();
+        if (playerMovement != null)
         {
-            PowerupActor manager = c.GetComponent<PowerupActor>();
+            playerMovement.SetExternalVelocityModifier(Vector3.one);
+        }
+    }
 
-            if (manager != null && !manager.GetDoubleSpeed())
-            {
-                manager.GoFaster();
-                AudioManager.m_Instance.PlaySound("MAGIC_Powerup", 1.0f, 1.0f);
-                Destroy(gameObject);
-            }
+    public override void OnPowerUpActivated()
+    {
+        PlayerMovement playerMovement = GetComponent<PlayerMovement>();
+        if (playerMovement != null)
+        {
+            playerMovement.SetExternalVelocityModifier(ComputeVelocityModifier());
```

Comment:
mmm I still think that the behaviour or method should be in the power up itself, so that when we create a new powerup we don't need to create a new method in player movement?

Reply:
This is the powerup itself isn't it? I'm just modifying the player variables here